tensorflow_probability/python/distributions/empirical.py

```diff
@@ -84,38 +84,33 @@ def __init__(self,
       name: Python `str` name prefixed to Ops created by this class.
     """
     parameters = locals()
-    with tf.name_scope(name, values=[params]):
-      if (params is None):
-        raise ValueError("Params is missing.")
-      with tf.control_dependencies([]):
-        self._params = tf.convert_to_tensor(params, name="params")
-        try:
-          self._n_sample = tf.shape(self._params)[0]
-        except:  # scalar params
-          self._n_sample = tf.constant(1)
+    with tf.name_scope(name, values=[samples]):
+      self._samples = tf.convert_to_tensor(samples, name="samples")
+      samples_shape_static = self._samples.shape.with_rank_at_least(1)
+      self._num_sample = tf.shape(self._samples)[0]
```

Comment:
Since `num_samples` is derived from `samples`, I don't think there's a reason to treat it as a separate graph parent. I'd just pass in `graph_parents=[self._samples]`, which should solve this issue.

Reply:
fixed.